ansible_collections/arista/avd/plugins/filter/ipaddr.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import ipaddress
+from jinja2.runtime import Undefined
+
+
+class FilterModule(object):
+
+    def net(self, val, version=None):
+        if isinstance(val, int) or '/' not in val:
+            return
+        try:
+            network_address = ipaddress.ip_network(val, strict=True)
+            if isinstance(version, int) and network_address.version != version:
+                return
+            return network_address
+        except ValueError:
+            return
+
+    def subnet(self, val, version=None):
+        if self.validate(val, version):
+            return ipaddress.ip_network(val, strict=False)
+        return
+
+    def netmask(self, val, version=None):
+        valid_subnet = self.subnet(val, version)
+        if valid_subnet:
+            return valid_subnet.netmask
+        return
+
+    def prefix(self, val, version=None):
+        valid_subnet = self.subnet(val, version)
+        if valid_subnet:
+            return valid_subnet.prefixlen
+        return
+
+    def size(self, val, version=None):
+        valid_subnet = self.subnet(val, version)
+        if valid_subnet:
+            return 2**(valid_subnet.max_prefixlen - valid_subnet.prefixlen)
+        return
+
+    def network(self, val, version=None):
+        valid_subnet = self.subnet(val, version)
+        if valid_subnet:
+            return valid_subnet.network_address
+        return
+
+    def address(self, val, version=None):
+        try:
+            if isinstance(val, str) and '/' in val:
+                try:
+                    if ipaddress.ip_network(val, strict=True):
+                        return
+                except ValueError:
+                    ip = ipaddress.ip_interface(val).ip
+            else:
+                ip = ipaddress.ip_address(val)
+            if isinstance(version, int) and ip.version != version:
+                return
+            return ip
+        except ValueError:
+            return
+
+    def validate(self, val, version=None):
+        try:
+            if isinstance(val, str) and '/' in val:
+                ip = ipaddress.ip_interface(val)
+            else:
+                ip = ipaddress.ip_address(val)
+            if isinstance(version, int) and ip.version != version:
+                return
+            return ip
+        except ValueError:
+            return
+
+    def defaultattr(self, input_val, ip_address):
+        try:
+            input_val_int = int(input_val)
+            if self.size(ip_address) == 1:
+                return ip_address
+            else:
+                max_ips = self.size(ip_address)
```

Comment:
```suggestion
            max_ips = self.size(ip_address)
            if max_ips == 1:
                return ip_address
            else:
```

one less computation